### PR TITLE
Improve lang#julia layer

### PR DESCRIPTION
**Pull request by wsdjeg**
### PR Prelude

Thank you for working on SpaceVim! :)

Please complete these steps and check these boxes before filing your PR:

- [x] I have read and understood SpaceVim's [CONTRIBUTING](https://github.com/SpaceVim/SpaceVim/blob/master/CONTRIBUTING.md) document.
- [x] I have read and understood SpaceVim's [CODE_OF_CONDUCT](https://github.com/SpaceVim/SpaceVim/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I have updated the [Following-HEAD](https://github.com/SpaceVim/SpaceVim/blob/master/wiki/en/Following-HEAD.md) page for this PR.
- [x] I understand my PR may be closed if it becomes obvious I didn't actually perform all of these steps.

### Why this change is necessary and useful?

julia has released v1.0, and we should improve julia support in SpaceVim.

- [x] Add English version lang#julia layer page 
- [x] Add Chinese version lang#julia layer page 
- [x] Add code runner
- [x] Add repl support
- [x] Add syntax/indent file
- [x] Add lsp support

### CODE REVIEW

**Comment by wsdjeg** [comment]
@jingpengw Hi, I just see your email on julia layer, and I just make some changes on this layer, can you help to review this code? BYW we are calling for code owner for language layer, would you like to join us? for more info, you can read #2002

File: docs/cn/layers/lang/julia.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+这一模块为 SpaceVim 提供了 julia 开发支持，包括代码补全、语法检查、以及代码格式化等特性。
+
+## 功能特性
+
+该模块主要包括插件 [julia-vim](https://github.com/JuliaEditorSupport/julia-vim)， 该插件提供：
+
+- 语法高亮
+
+
+同时，该模块还未 elixir 开发提供了交互式编程和调试等功能。
```
Comment (xiuliren) [inline docs/cn/layers/lang/julia.md]: typo: 还为

File: docs/layers/language-server-protocol.md [hunk 2]
```diff
@@ -79,6 +79,14 @@ npm install -g javascript-typescript-langserver
 pip install --user python-language-server
 ```
 
+**julia:**
+
+The `LanguageServer` package must be installed in Julia (0.6 or greater), i.e.
+
+```sh
+julia> Pkg.clone("https://github.com/JuliaEditorSupport/LanguageServer.jl")
```
Comment (xiuliren) [inline docs/layers/language-server-protocol.md:87]: with new package system in Julia 0.7 and above, we have a package mode in Julia REPL. 
in REPL, hit "]" to enter the package management mode, then `add LanguageServer` to install the package.

File: docs/layers/language-server-protocol.md [hunk 3]
```diff
@@ -97,6 +105,7 @@ default language server commands:
 | language     | server command                                   |
 | ------------ | ------------------------------------------------ |
 | `javascript` | `['javascript-typescript-stdio']`                |
+| `typescript` | `['typescript-language-server', '--stdio']`      |
```
Comment (xiuliren) [inline docs/layers/language-server-protocol.md:111]: this is adding typescript support?

File: autoload/SpaceVim/layers/lang/julia.vim [hunk 4]
```diff
@@ -9,32 +9,47 @@
...
 endfunction
 
 
 function! SpaceVim#layers#lang#julia#config() abort
+  " registe code runner commmand for julia
+  call SpaceVim#plugins#runner#reg_runner('julia', 'julia %s')
+  " registe REPL command and key bindings for julia
+  call SpaceVim#plugins#repl#reg('julia', 'julia')
+  let g:_spacevim_mappings_space.l.s = {'name' : '+Send'}
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','s', 'i'],
```
Comment (xiuliren) [inline autoload/SpaceVim/layers/lang/julia.vim:31]: the `i` is a abbreviation of `interactive`?

**Review by xiuliren — COMMENTED** [review]
I am not familiar with vim language, could only add some comments on Julia itself.

**Comment by xiuliren** [comment]
Dear Shidong,

I am happy to be one of the Julia layer owners, but not sure that I am
qualified to be or not. I am not familiar with vim language, but would like
to learn a little bit. I probably will still need your review and help.

Jingpeng

File: docs/layers/language-server-protocol.md (hunk 2 above)
Comment (wsdjeg) [reply docs/layers/language-server-protocol.md:87]: The installation step is from LanguagerServer's wiki,  ok, I will add this.

File: docs/layers/language-server-protocol.md (hunk 3 above)
Comment (wsdjeg) [reply docs/layers/language-server-protocol.md:111]: yeah, ts lsp support has been added in old verison, but the doc is missing.

File: autoload/SpaceVim/layers/lang/julia.vim (hunk 4 above)
Comment (wsdjeg) [reply autoload/SpaceVim/layers/lang/julia.vim:31]: yeah, these repl command are same as other language layer. also in spacemacs.

**Pull request merged**